src/App.tsx

```diff
@@ -24,10 +25,16 @@ export class App extends React.Component<{}, {}> {
     this.setState({ selectedGood: '' });
   };
 
-  selectBtn = (item: string) => {
+  selectButton = (item: string) => {
     const { selectedGood } = this.state;
+    const isSelected = selectedGood === item;
 
-    if (selectedGood !== item) {
+    // this.setState({
+    //   selectedGood: isSelected
+    //     ? item
+    //     : '',
+    // });
```

Comment:
I didn't get it(

Reply:
You can check what to set directly in setState using ternary or logic operators

```
 this.setState({
       selectedGood: isSelected
        ? item
        : '',
 });
    
```

For example you want to set user name to state if user has it and null if don't:
```
this.setState({ name: user.name || null })
```

Instead of : 
```
if(user.name) {
   this.setState({  name: user.name });
} else {
   this.setState({ name: null });
}
```

Reply:
Ohh, I am stupid)) I just forgot add '!' before 'isSelected'
Thank you!